### PR TITLE
UI Cleanup

### PR DESCRIPTION
**Pull request by atila-crabnebula**
fixes DT-6

this PR changes all views with a few UX, performance, and accessibility improvements.

No features were impacted, just slightly modified for better experience.

### CODE REVIEW

File: clients/web/src/components/disconnect-button.tsx [hunk 1]
```diff
@@ -9,13 +9,16 @@ export function DisconnectButton() {
...
       id="close"
-      class="border-gray-700 border rounded-md p-1 inline-flex items-center justify-center text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-indigo-500"
+      class="border-slate-800 border w-6 hover:w-[8em] focus:w-[8em] transition-all duration-500 ease-in-out  rounded-md p-1 inline-flex items-center justify-end text-slate-200  focus:outline-none focus:ring-2 focus:ring-inset focus:ring-indigo-500 group relative overflow-x-hidden"
       onClick={() => {
         connectionStore.abortController.abort();
         goto("/");
       }}
     >
       <span class="sr-only">Close connection</span>
+      <span class="group-hover:inline-block group-focus:inline-block hidden absolute leading-[0] px-2 right-6 text-slate-200">
```
Comment (johann-crabnebula) [inline clients/web/src/components/disconnect-button.tsx:19]: Fixing the spacing would be nice:

distance

**Review by johann-crabnebula — CHANGES_REQUESTED** [review]

File: clients/web/src/components/tauri/configuration-view.tsx [hunk 2]
```diff
@@ -40,16 +41,24 @@ export function ConfigurationView() {
...
       <Show when={params.config && !params.selected && config()}>
         <h1 class="text-3xl">{config()?.label}</h1>
         <ConfigurationErrors error={config()?.error} />
       </Show>
-      <Show when={tabWithKeys(tab())}>
+      <Show
+        when={tabWithKeys(tab())}
+        fallback={
+          <div class="h-full grid gap-4 text-center content-center justify-center items-center border-l p-4 border-gray-800">
+            <Heading>No File Selected</Heading>
```
[the remark this answers is not in the document]
Comment (johann-crabnebula) [reply clients/web/src/components/tauri/configuration-view.tsx:53]: Not quite sure about this one, but files doesn't quite cut it

**Comment by johann-crabnebula** [comment]
Ah and these look a bit weird:

chrome_R3gk7yZbxF

chrome_4GtlfapTkB

File: clients/web/src/components/tauri/configuration-tooltip.tsx [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import { Tooltip } from "@kobalte/core";
 import { useConfiguration } from "~/components/tauri/configuration-context";
 import SolidMarkdown from "solid-markdown";
-import { Show, For, Switch, Match } from "solid-js";
+import { Show, For, Switch, Match, onMount } from "solid-js";
```
[the remark this answers is not in the document]
Comment (atila-crabnebula) [reply clients/web/src/components/tauri/configuration-tooltip.tsx:4]: thanks! Just noticed this in the CI!

File: clients/web/src/views/dashboard/sources.tsx [hunk 4]
```diff
@@ -11,6 +11,7 @@ export default function Sources() {
   return (
     <SplitPane
       defaultPrefix="sources"
+      gutterSize={50}
```
[the remark this answers is not in the document]
Comment (atila-crabnebula) [reply clients/web/src/views/dashboard/sources.tsx:14]: definitely not. I was working around a browser quirk... gotta fix it, thanks for catching this!

File: clients/web/src/views/dashboard/sources.tsx
[the remark this answers is not in the document]
Comment (atila-crabnebula) [reply clients/web/src/views/dashboard/sources.tsx:1]: browser quirk!
https://github.com/crabnebula-dev/devtools/pull/216#discussion_r1466609871

File: clients/web/src/components/disconnect-button.tsx (hunk 1 above)
Comment (atila-crabnebula) [reply clients/web/src/components/disconnect-button.tsx:19]: good eye! Thanks!

File: clients/web/src/components/errors/general-error.tsx [hunk 5]
```diff
@@ -1,7 +1,9 @@
 export function GeneralError() {
   return (
     <>
-      <h1 class="text-red-400 font-semibold text-6xl">Irrecoverable Error</h1>
+      <h1 class="text-red-400 font-semibold text-6xl">
+        Oops! We can come back...{" "}
```
[the remark this answers is not in the document]
Comment (atila-crabnebula) [reply clients/web/src/components/errors/general-error.tsx:5]: good catch! I'll fix it!

File: clients/web/src/components/tauri/configuration-view.tsx [hunk 6]
```diff
@@ -40,53 +40,54 @@ export function ConfigurationView() {
...
-        <MissingConfigurationDialog config={params.config} />
-      </Show>
-      <Show when={params.config && config() && params.selected && !tab()}>
-        <MissingConfigurationParameterDialog
-          config={params.config}
-          selectedParameter={params.selected}
-        />
-      </Show>
+    <div class="p-4 h-full">
+      <Switch>
```
Comment (johann-crabnebula) [inline clients/web/src/components/tauri/configuration-view.tsx:44]:
```suggestion
      <Switch
        fallback={
          <div class="h-full grid gap-4 text-center content-center justify-center items-center p-4">
            <Heading>No Configuration Selected</Heading>
            &larr; Use the sidebar to get started.
          </div>
        }
      >
```

File: clients/web/src/components/tauri/configuration-view.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
-import { Show, For, createEffect, Switch, Match } from "solid-js";
+import { For, createEffect, Switch, Match } from "solid-js";
 import { ConfigurationValue } from "./configuration-value";
 import { ConfigurationTooltip } from "./configuration-tooltip";
 import { generateDescriptions } from "~/lib/tauri/tauri-conf-lib";
```
Comment (johann-crabnebula) [inline clients/web/src/components/tauri/configuration-view.tsx:4]:
```suggestion
import { generateDescriptions } from "~/lib/tauri/tauri-conf-lib";
import { Heading } from "../heading";
```

Comment (johann-crabnebula) [reply clients/web/src/components/tauri/configuration-view.tsx:4]: I liked the fallback

File: clients/web/src/components/tauri/configuration-view.tsx (hunk 6 above)
Comment (johann-crabnebula) [reply clients/web/src/components/tauri/configuration-view.tsx:44]: I liked the fallback

File: clients/web/src/components/tauri/configuration-value/flags.tsx [hunk 8]
```diff
@@ -12,17 +12,19 @@ export function Flags(props: { key: string; value: boolean | string }) {
     return lineNumber > 0;
   };
   return (
-    <span class="text-neutral-400 pr-2 font-mono">
+    <span class="text-neutral-400 pl-2 text-sm">
       <Switch>
         <Match
           when={isDefaultValue(localSchema(), props.value) && isInConfig()}
         >
           Set Default
```
Comment (johann-crabnebula) [inline clients/web/src/components/tauri/configuration-value/flags.tsx:20]:
```suggestion
          (set default)
```

Comment (johann-crabnebula) [reply clients/web/src/components/tauri/configuration-value/flags.tsx:20]: Or did you resolve this on purpose?

File: clients/web/src/styles.css [hunk 9]
```diff
@@ -6,8 +6,8 @@
 @layer base {
   /** fonts */
   :root {
-    --header-height: 69px;
-    --toolbar-height: 28px;
+    --header-height: 90px;
```
Comment (johann-crabnebula) [inline clients/web/src/styles.css:9]: This seems to create a gap between the border and the following tab content:

chrome_ZHlR3BAKhm

**Review by johann-crabnebula — CHANGES_REQUESTED** [review]

File: clients/web/src/components/tauri/configuration-value/flags.tsx (hunk 8 above)
Comment (atila-crabnebula) [reply clients/web/src/components/tauri/configuration-value/flags.tsx:20]: I believe I had committed it

**Review by johann-crabnebula — APPROVED** [review]

**Pull request merged**